stdlib/dynarray.mli

```diff
@@ -599,6 +599,19 @@ val reset : 'a t -> unit
     in the backing array at index [length a] or later.
 *)
 
+val unsafe_to_iarray : capacity:int -> ('a t -> unit) -> 'a iarray
+(** [unsafe_to_iarray ~capacity f] calls [f] on a new empty dynarray with the
+    given [capacity], then turns it into an immutable array without a copy,
+    when possible (i.e, the array is not a float array, or flat float arrays
+    are disabled).
+    When [f] returns, the dynarray's capacity must be exactly equal to its
+    length, otherwise the operation cannot be performed without a copy. If it
+    is not the case, [Invalid_argument] is raised. Note that the dynarray's
```

Comment:
I don't think this is a good behaviour. I want to be able to speculate on `capacity`. If the dynarray is not equal to its capacity, just make a copy.

Reply:
Unconditionally calling `fit_capacity` after `f` should do the trick

Reply:
I suspose you meant at the end of `f`. 

But why make it so complicated for the user ? This is not the spirit of a dynarray whose idea is that I should not worry about its size. 

I think this is a wrong and unergonomic default. If you worry about having exactly matched the `capacity` just check and raise yourself in `f`.

Reply:
I think @smuenzel meant that by calling `fit_capacity` after `f` in the implementation, we get the behaviour that you want, namely no copy if the capacity matches the length.

Reply:
Ah sorry for the misinterpretation I thought he was talking about client code.